test/typescript-validate.ts

```diff
@@ -0,0 +1,15 @@
+import * as Octokit from '../'
+
+// This code is NOT executed.
+
+// Test the TypeScript definition.
+export default async function() {
+  new Octokit() // Check empty constructor
+  const octo = new Octokit({}) // Check that all arguments are optional
+  const repo = await octo.repos.get({owner: 'octokit', repo: 'rest.js'})
```

Comment:
I would like the tests to be able to run when offline. Could you add mocking with nock, or get the instance like we do in the scenario tests, e.g.

https://github.com/octokit/rest.js/blob/bf2f550557bfed7499ef197d22c4c58f71754f02/test/scenarios/get-repository-test.js#L8-L14

Reply:
This code was not intended to be run (it does not match `'test/**/*-test.js'`). 

The "test" is just that `tsc` does not complain when compiling the `.ts` file; it would complain if, for example, you edited the file to say `repo.meta.linkThisFieldIsInvalid`.

Is there a better place for this file to make it clear that the generated `.js` file is not actually executed?

Reply:
ah okay that makes sense. Maybe an explanatory comment in the `test/typescript-validate.ts` would help? It is all kind of confusing ... do you know if is there a command option for `tsc` that does not create a js file, only checks for compilation errors?

Reply:
I included a small comment at the top of the file but it seems like it was overlooked. In the most recent commit I made it a bit bigger 😄

Reply:
I was very tired when I checked that. New comment is great, thank you :)